
The et driver assumes that the sk_buff->priority field is going to be < 8
and therefore uses it as an index into an 8 element array of transmit queues.
This is a valid assumption unless you use the CLASSIFY target. In such case,
netfilter uses the priority field to communicate the target class:
 10020 for class 1:20, 10040 for class 1:40, etc.

This causes an array overrun and the panic.

diff -uBp a/et/sys/etc.h b/et/sys/etc.h
--- a/et/sys/etc.h	2010-11-22 11:59:20.000000000 +0300
+++ b/et/sys/etc.h	2011-09-15 14:17:32.000000000 +0400
@@ -17,6 +17,7 @@
 #define _etc_h_
 
 #include <etioctl.h>
+#include <proto/802.1d.h>
 
 #define	MAXMULTILIST	32
 
@@ -193,9 +194,9 @@ typedef struct etc_info {
 static inline uint32
 etc_up2tc(uint32 up)
 {
-	extern uint32 up2tc[];
+	extern uint32 up2tc[NUMPRIO];
 
-	return (up2tc[up]);
+	return (up <= MAXPRIO ? up2tc[up] : 0);
 }
 
 static inline uint32
-- 
